### PR TITLE
Multi-process oplog reader

### PR DESCRIPTION
**Pull request by will123195**
- [x] normal operation tests

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+            delete this.oplogs[connStr];
+          });
+      }
+    });
+    newConnStrings.forEach(newConnStr => {
+      const exists = Object.keys(this.oplogs).includes(newConnStr);
+      if (!exists) {
+        // start reading the oplog of a new shard
+        console.log('new oplog:', newConnStr);
+        // TODO: do we need to start the tail at a time in the past?
```
Comment (will123195) [inline src/index.js]: I think it's ok to start tailing "now" in the case that a new shard is added. Thoughts?

Comment (wwbenomathew) [reply src/index.js]: Im wondering if we should do now-1 hour, for safety reasons. What if there's a write just a millisecond before now?

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+      });
+  }
+
+  setMembersOfReplSet(db) {
+    return db.admin().replSetGetStatus()
+      .then(info => {
+        const replSetName = info.set;
+        this.replSets[replSetName] = this.replSets[replSetName] || {};
+        info.members.forEach(member => {
+          if (!member.health) return;
```
Comment (wwbenomathew) [inline src/index.js]: should we be returning anything?

File: src/index.js (hunk 1 above)
Comment (will123195) [reply src/index.js]: I'll start tailing from the beginning of the oplog if this replset has no `lastOpTime` in redis.

File: src/index.js (hunk 2 above)
Comment (will123195) [reply src/index.js]: no, we're ignoring unhealthy members

**Comment by will123195** [comment]
closing in favor of #4

**Pull request closed without merging**